File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.java
class Solution {
    public int largestRectangleArea(int[] heights) {
        int n = heights.length;
        int max =0;
        
        for(int i=0; i<n;i++){

            int h = heights[i];

            int r = i+1;

            while(r<n&& heights[r]>=h){
                r++;
            }
            int l = i;

            while(l>=0&& heights[l]>=h){
                l--;
            }
            int ar = h*(r-l-1);

            max = Math.max(max,ar);
        }

        return max;
    }
}

File: Data Structures & Algorithms/longest-common-subsequence/submission-0.swift
class Solution {
    func longestCommonSubsequence(_ text1: String, _ text2: String) -> Int {

        let arr1 = Array(text1)
        let arr2 = Array(text2)

        var dp = Array(repeating:Array(repeating:0, count:arr2.count+1), count: arr1.count+1)

        var ans = 0


        for i in 1 ... arr1.count {
            for j in 1 ... arr2.count {

                if arr1[i-1] == arr2[j-1] {

                    dp[i][j] = dp[i-1][j-1] + 1
                    ans = max(dp[i][j], ans)

                } else {
                    dp[i][j] = max(
                        dp[i - 1][j], // skip text1 char
                        dp[i][j - 1]  // skip text2 char
                    )
                }

            }

        }

        return dp[arr1.count][arr2.count]

    }
}
